pkg/storagecapabilities/storagecapabilities.go

```diff
@@ -269,17 +269,17 @@ var storageClassToProvisionerKeyMapper = map[string]func(sc *storagev1.StorageCl
 		return "pxd.openstorage.org"
 	},
 	"kubernetes.io/portworx-volume": func(sc *storagev1.StorageClass) string {
-		val := sc.Parameters["shared"]
-		if val == "true" {
-			return "kubernetes.io/portworx-volume/shared"
+		val := sc.Parameters["sharedv4_mount_options"]
+		if strings.Contains(val, "vers=3.0") && strings.Contains(val, "nolock") {
+			return "kubernetes.io/portworx-volume/nfs"
 		}
 		return "kubernetes.io/portworx-volume"
 	},
 	"pxd.portworx.com": func(sc *storagev1.StorageClass) string {
-		// https://docs.portworx.com/portworx-install-with-kubernetes/storage-operations/csi/volumelifecycle/#create-shared-csi-enabled-volumes
-		val := sc.Parameters["shared"]
-		if val == "true" {
-			return "pxd.portworx.com/shared"
+		// https://docs.portworx.com/portworx-enterprise/operations/operate-kubernetes/storage-operations/manage-kubevirt-vms.html#create-a-storageclass
```

Comment:
How are upgrades handled? Will this change also update existing StorageProfiles which have incorrect capabilities ordering?

Reply:
CDI upgrade will reconcile all `StorageProfiles` and update  to the fixed `claimPropertySets`. You can also delete a `StorageProfile` (e.g. in case you edited its spec and want to drop the changes) and it will be recreated correctly.